### DISCUSSION TITLE
DataGrid.Styles Converter not firing when editing cell directly or from code

### DISCUSSION THREAD

**Discussion by davi2td** [AvaloniaUI/Avalonia#12321] [Q&A] [answered]
**Describe the bug**
-     Test Avalonia Datagrid.Styles converter Intial Load and adding rows works
-     Editing any firstname cell in grid (or hitting change prop button) and changing to "Neil" does not fire a color change to green
-     Sorting on any column does force refresh
I'm probably just getting something wrong?
**To Reproduce**
Probably best to use my test solution on github: [AvaloniaTestColorConverter](https://github.com/davi1td/AvaloniaTestColorConverter/tree/master)

1. Edit any name in First Name column to "Neil", color does not change to green, CellColorConverter converter does not fire
2. Sort any column and then all First Name cells named "Neil" update

**Expected behavior**
Any changes to cell in grid , via code or direct edit fires converter.

**Desktop (please complete the following information):**
- OS: Windows, Mac, Linux (State distribution)]
- Version 11

**Additional context**

**Comment by timunie** [comment, marked as answer]
@davi1td you bind to the `DataContext` which is `Person`. But the UI gets notified that `Person.FirstName` has changed, not `Person`. So the UI will not call the converter. If you change the Binding like that, it should work:

```xml
<Setter Property="Background" Value="{Binding FirstName, Converter={StaticResource CellColorConverter}}" />
```

and update converter:
```cs
public object? Convert(object? value, Type targetType, object? parameter, CultureInfo culture)
{
  if (value is not string firstName) return new SolidColorBrush(Colors.Blue);
  return firstName switch
  {
      "Neil" => new SolidColorBrush(Colors.Green),
      "Buzz" => new SolidColorBrush(Colors.Yellow),
      _ => new SolidColorBrush(Colors.Purple)
  };
}
```

Hint: If you wonder why it works while sorting: The containers are refreshed and thus they pull the binding again. TwoWay is not needed here btw as you do not expect user input here. 

Hope this helps.

**Reply by davi2td** [reply]
You rock! Can't thank you enough. I take it in the future I should post in discussion? Sorry for posting in issues ... ;)

**Reply by timunie** [reply]
Depends. if you really think it's a bug, you can file an issue. If you think it might be more a question, use Disussions or Telegram chat https://t.me/Avalonia

> Sorry for posting in issues ... ;) 

All fine, you're welcome :-)

**Reply by davi2td** [reply]
One more question, how would I make a decision in the CellColorConverter based another cell? For example, whether or not IsFictitious is true or false. I started to go down the path of MultiBinding or ConverterParameter but for the life of me can't get a clue ;)

**Reply by timunie** [reply]
Yeah, MultiBinding or add another property into Person. You can manually raise property changed for that property. 

MultiConverter: https://github.com/AvaloniaUI/Avalonia.Samples/tree/main/src/Avalonia.Samples/MVVM/ValueConversionSample#solution-2--using-imultivalueconverter

**Reply by davi2td** [reply]
I'll look, once again thanks!

**Comment by timunie** [comment]
As this makes your sample work (filed you a PR), I'll convert this into a discussion now.
